### PR TITLE
Feature/garbage-collection

### LINKED ISSUE #2: Garbage Collection for unused Guilds
- When Bot leaves channel, internal data for this channel hast to be removed from the database
- Scheduler Tasks must be stopped

### PR DESCRIPTION
**Pull request by SRTigers98**
Created a service with a scheduled method which runs once every week.
The method reads all guild configurations from the database and checks if there are guilds in the database the bot is not enrolled in anymore.
Such stale guilds will be removed from the database including all associated birthdays.

- Fixes #2

### CODE REVIEW

File: src/main/kotlin/io/github/srtigers98/birthdaydiscordbot/application/service/CleanupService.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+class CleanupService(
+  private val restClient: RestClient,
+  private val guildConfigService: GuildConfigService,
+) {
+
+  private val log = LoggerFactory.getLogger(CleanupService::class.java)
+
+  /**
+   * Checks once per week the current configured guilds.
+   * If a guild is configured the bot is no member anymore the configuration and birthdays associated with that guild will be deleted.
```
Comment (boomer41) [inline src/main/kotlin/io/github/srtigers98/birthdaydiscordbot/application/service/CleanupService.kt:24]: > Checks all guilds in the database.
> If a guild is saved in the database, but the bot is no longer a member of this guild, the respective configuration and all birthdays are deleted.

File: src/main/kotlin/io/github/srtigers98/birthdaydiscordbot/application/service/CleanupService.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    val enrolledGuildIds = restClient.user.getCurrentUserGuilds()
+      .map { it.id.toString() }
+
+    val staleGuilds = configuredGuilds
+      .filter { !enrolledGuildIds.contains(it.guildId) }
+      .toTypedArray()
+
+    guildConfigService.deleteGuildConfig(*staleGuilds)
+
+    log.info("Deleted ${staleGuilds.size} stale guild configuration(s)! -> ${staleGuilds.map { it.guildId }}")
```
Comment (boomer41) [inline src/main/kotlin/io/github/srtigers98/birthdaydiscordbot/application/service/CleanupService.kt:40]: How does `${staleGuilds.map { it.guildId }}` produce a string containing all guild ids?
How does that string look like?

**Review by boomer41 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/github/srtigers98/birthdaydiscordbot/application/service/CleanupService.kt (hunk 2 above)
Comment (SRTigers98) [reply src/main/kotlin/io/github/srtigers98/birthdaydiscordbot/application/service/CleanupService.kt:40]: The statement creates an array of strings which are represented as [id1,id2,id3]

File: src/main/kotlin/io/github/srtigers98/birthdaydiscordbot/application/service/CleanupService.kt (hunk 1 above)
Comment (SRTigers98) [reply src/main/kotlin/io/github/srtigers98/birthdaydiscordbot/application/service/CleanupService.kt]: Updated javadoc

**Review by boomer41 — APPROVED** [review]

**Pull request merged**